apps/web/src/components/Settings/ChoosePicture.tsx

```diff
@@ -0,0 +1,161 @@
+                    : 'border-transparent opacity-50 hover:bg-indigo-500/[0.12] rounded-tl-lg rounded-tr-lg'
+                )
+              }
+            >
+              Upload
+            </Tab>
+            <Tab
+              className={({ selected }) =>
+                clsx(
+                  'py-2 border-b-2 text-sm focus:outline-none w-full',
+                  selected
+                    ? 'border-indigo-500 opacity-100'
+                    : 'border-transparent opacity-50 hover:bg-indigo-500/[0.12] rounded-tl-lg rounded-tr-lg'
+                )
+              }
+            >
+              NFT's
+            </Tab>
+          </Tab.List>
+          <Tab.Panels>
+            <Tab.Panel className="no-scrollbar h-96 focus:outline-none flex flex-col justify-center py-10">
+              <div className="flex justify-center">
+                <label
+                  htmlFor="choosePfp"
+                  className="border border-indigo-500 bg-indigo-500 px-8 py-4 rounded-full text-white cursor-pointer"
+                >
+                  Choose image
+                </label>
+                <input
+                  id="choosePfp"
+                  type="file"
+                  accept=".png, .jpg, .jpeg, .svg, .gif"
+                  className="hidden w-full"
+                  onChange={onChooseImage}
+                />
+              </div>
+            </Tab.Panel>
+            <Tab.Panel className="no-scrollbar h-96 overflow-y-auto focus:outline-none py-5">
+              {!error && !loading && (
+                <>
+                  <div className="grid grid-cols-3 gap-5">
+                    {collectedNFTs.map((nft: Nft) => {
+                      return (
+                        <img
+                          key={nft.contentURI}
+                          src={imageCdn(
+                            nft.originalContent?.uri
+                              ? sanitizeIpfsUrl(nft.originalContent?.uri)
+                              : `${STATIC_ASSETS}/images/placeholder.webp`,
+                            'thumbnail'
+                          )}
+                          className="rounded-xl cursor-pointer h-20 w-36"
+                          alt={nft.name}
+                          onClick={() =>
+                            setNFTAvatar(
+                              nft.contractAddress,
+                              nft.tokenId,
+                              nft.chainId
+                            )
+                          }
+                        />
+                      )
+                    })}
+                  </div>
+                  {pageInfo?.next &&
+                    collectedNFTs.length !== pageInfo?.totalCount && (
+                      <span ref={observe} className="flex justify-center mt-5">
+                        <Loader />
+                      </span>
+                    )}
+                </>
+              )}
+              {loading && <TimelineShimmer />}
```

Comment:
move the shimmer above the map nfts and timeline shimmer looks good for a square image grid. we can use just middle spinner if needed